### PR TITLE
Use --with-artifacts instead of --no-download

### LINKED ISSUE #4: Change --no-download option into --with-artifacts
See https://github.com/marco-c/grcov/issues/20#issuecomment-296778854.

### PR DESCRIPTION
**Pull request by rhcu**
Fixes #4

### CODE REVIEW

File: codecoverage.py [hunk 1]
```diff
@@ -208,24 +208,22 @@ def main():
     parser.add_argument('branch', action='store', nargs='?', default=default_branch, help='Branch on which jobs ran')
     parser.add_argument('commit', action='store', nargs='?', default=default_commit, help='Commit hash for push')
     parser.add_argument('--grcov', action='store', nargs='?', help='Path to grcov')
-    parser.add_argument('--no-download', action='store_true', help='Use already downloaded coverage files')
-    parser.add_argument('--no-grcov', action='store_true', help='Use already generated grcov output (implies --no-download)')
+    parser.add_argument('--with-artifacts', action='store', nargs='?', help='Path to already downloaded coverage files')
+    parser.add_argument('--no-grcov', action='store_true', help='Use already generated grcov output (implies --with-artifacts PATH_TO_ARTIFACTS)')
```
Comment (marco-c) [inline codecoverage.py]: We should update the comment, as this no longer implies `--with-artifacts`

File: codecoverage.py [hunk 2]
```diff
@@ -208,24 +208,22 @@ def main():
...
     args = parser.parse_args()
 
-    if args.no_grcov:
-        args.no_download = True
-
-    if (args.branch is None) != (args.commit is None) and not args.no_download:
-        parser.print_help()
-        return
+    if not args.with_artifacts:
+        if (args.branch is None) != (args.commit is None) or args.no_grcov:
```
Comment (marco-c) [inline codecoverage.py]: If there's no `with_artifacts` option, you should still be able to use `no_grcov`, so you can remove `or args.no_grcov`

File: codecoverage.py [hunk 3]
```diff
@@ -238,7 +236,10 @@ def main():
             download_grcov()
             grcov_path = './grcov'
 
-        generate_report(grcov_path, 'lcov', 'output.info')
+        if args.with_artifacts:
```
Comment (marco-c) [inline codecoverage.py]: You can do:
```
generate_report(grcov_path, 'lcov', 'output.info', args.with_artifacts)
```

This way if args.with_artifact is None, generate_report should use the default 'ccov-artifacts'

Comment (marco-c) [reply codecoverage.py]: Right, forget my suggestion.

File: codecoverage.py [hunk 4]
```diff
@@ -208,24 +208,22 @@ def main():
...
     parser.add_argument('--ignore', action='store', nargs='+', help='List of test suites to ignore (by default \'talos\' and \'awsy\'). E.g. \'mochitest\', \'mochitest-chrome\', \'gtest\', etc.')
     args = parser.parse_args()
 
-    if args.no_grcov:
-        args.no_download = True
-
-    if (args.branch is None) != (args.commit is None) and not args.no_download:
-        parser.print_help()
-        return
+    if not args.with_artifacts:
```
Comment (marco-c) [inline codecoverage.py]: We shouldn't download the artifacts also when `args.no_grcov`

Comment (rhcu) [reply codecoverage.py]: Won't this mean that `args.no_grcov` implies `with_artifacts`?

Comment (marco-c) [reply codecoverage.py]: Not exactly, as `with_artifacts` requires an argument.

**Pull request merged**